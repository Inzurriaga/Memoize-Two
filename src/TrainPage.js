import React from "react"
import QuestionContainer from "./QuestionContainer"
import QuestionElement from "./QuestionElement"

export default function TrainPage(props) {
    return (
        <div>
            {
                !props.Questiontoggle ?
                    <div>
                        <QuestionContainer
                            user={props.user}
                            toggleToQuestion={props.toggleToQuestion}/>
                    </div> :
                    <QuestionElement question={props.selectedQuestion}
                                    users={props.users}
                                    user={props.user} />
            }
        </div>
    )
}