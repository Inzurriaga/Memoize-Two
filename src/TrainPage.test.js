import React from 'react';
import { shallow } from 'enzyme';
import TrainPage from "./TrainPage"

const mockUser = {"name":"gabriel","password":"gabriel","completed":[1,2,5]};
const mockUsers = [{"name":"duy","password":"duy","completed":[1,2,5]},{"name":"gabriel","password":"gabriel","completed":[1,2,5]},{"name":"isaac","password":"isaac","completed":[1,2]}];
const mockToggle = false;
const mockToggleFunction = jest.fn();
const mockSelectedQuestion = { title: "number return 5", type: "number", instructions: "return the number 5", answer: 5, startup: "function number() {\n//code here \n} \n\n number()", id: 1 };


describe("TrainPage", ()=> {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(
        <TrainPage users={mockUsers}
                    user={mockUser}
                    Questiontoggle={mockToggle}
                    toggleToQuestion={mockToggleFunction}
                    selectedQuestion={mockSelectedQuestion}/> 
      )
    })

    it("should match Snap Shot", () => {
      expect(wrapper).toMatchSnapshot()
    })

    it("should which to the question element snap shot", () => {
      const mockToggle = true;
      wrapper = shallow(
        <TrainPage users={mockUsers}
                    user={mockUser}
                    Questiontoggle={mockToggle}
                    toggleToQuestion={mockToggleFunction}
                    selectedQuestion={mockSelectedQuestion}/> 
      )
    })
})