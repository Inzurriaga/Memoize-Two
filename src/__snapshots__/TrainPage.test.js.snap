// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrainPage should match Snap Shot 1`] = `
<div>
  <div>
    <QuestionContainer
      toggleToQuestion={[MockFunction]}
      user={
        Object {
          "completed": Array [
            1,
            2,
            5,
          ],
          "name": "gabriel",
          "password": "gabriel",
        }
      }
    />
  </div>
</div>
`;
